src/main/java/StringCalculator.java

```diff
@@ -0,0 +1,34 @@
+import number.Number;
+import splitter.Splitter;
+import splitter.SplitterManager;
+import util.StringUtil;
+
+import java.util.Arrays;
+
+public class StringCalculator {
+
+    public int calculate(final String value) {
+        if (StringUtil.isEmpty(value)) {
+            return Number.ZERO_VALUE;
+        }
+        Splitter splitter = SplitterManager.of().matchedSplitter(value);
+        return sum(splitter.split(value));
+    }
+
+    private int sum(final String[] numbers) {
+        if (this.isEmptyNumbers(numbers)) {
+            return Number.ZERO_VALUE;
+        }
+        return Arrays.stream(numbers)
+                .map(Number::of)
+                .reduce(Number.ZERO, Number::plus)
+                .getValue();
+    }
+
+    private boolean isEmptyNumbers(final String[] numbers) {
+        if (numbers.length == 1) {
+            return numbers[0].isEmpty();
+        }
+        return false;
```

Comment:
```suggestion
        return numbers == EMPTY_ARRAY;
```
이런 방식은 어떤가요?
여기는 처음 봤을때 어떤건지 확 느낌이 안오더라구요

Reply:
아 이게 아니구나.......??ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ

Reply:
좀 더 보고 다시 달게요!

Reply:
저도 이거 때문에 엄청 고민을 했는데... validate 를 한 이유가

- [x] 커스텀 구분자 형식의 문자열 중 숫자값을 빈 문자열로 입력한 경우 0을 반환한다. (예 : “//:\n” => 0) 

이 요구사항 때문인데, 이 경우가 EMPTY_ARRAY를 의미하는 게 아니라 String 배열에서 빈 문자열이 하나가 있는 상황입니다 😭

Reply:
```
private static final String EMPTY_OPERAND_REGEX = "//(.)\n";
private static final Pattern emptyOperandPattern = Pattern.compile(EMPTY_OPERAND_REGEX);

@Override
public String[] split(final String value) {
    if (value.isEmpty() || isMatchedEmptyOperandPattern(value)) {
        return EMPTY_ARRAY;
    }

    Matcher matcher = customPattern.matcher(value);
    if (!matcher.matches()) {
        throw new IllegalArgumentException();
    }

    String customDelimiter = this.convertEscapedString(matcher.group(INDEX_OF_DELIMITER));
    return matcher.group(INDEX_OF_VALUE).split(customDelimiter);
}

private boolean isMatchedEmptyOperandPattern(final String value) {
    return emptyOperandPattern.matcher(value)
                .matches();
}
```
- 이렇게 변경하니까 의도도 확실하게 보이고 훨씬 깔끔해진 거 같네요ㅎㅎㅎ 👍 감사합니다~